### DISCUSSION TITLE
Should poetry build respect the lockfile?

### DISCUSSION THREAD

**Discussion by craigjbass** [python-poetry/poetry#7912] [Q&A] [answered]
If poetry install results in dev dependencies being installed, that pull in a transient dependency into the virtual environment

However the "production" dependency graph is different - the transient dependency could be installed at a newer version.

Would we expect the `poetry build` command to install the transient dependency to respect the lock file, or to install the newer version as specified by the production dependency graph?

**Comment by dimbleby** [comment]
`poetry build` doesn't install anything at all, there's no question of it choosing a version.

**Reply by craigjbass** [reply]
Thanks!

**Comment by sbrunner** [comment, marked as answer]
`poetry build` create a package that resect what's in the `pyproject.toml` the `poetry.lock` isn't used.

**Reply by craigjbass** [reply]
Thanks!

**Comment by dimbleby** [comment]
building a package with pinned dependencies is #2778

**Reply by craigjbass** [reply]
Thanks!
